src/libraries/iKin/CMakeLists.txt

```diff
@@ -27,19 +27,22 @@ if(ICUB_USE_IPOPT)
                      include/iCub/iKin/iKinSlv.h)   
 endif()
 
-source_group("Source Files" FILES ${folder_source})
-source_group("Header Files" FILES ${folder_header})
-
-include_directories(${PROJECT_SOURCE_DIR}/include
-                    ${ctrlLib_INCLUDE_DIRS})
-
 if(ICUB_USE_IPOPT)
-   include_directories(${IPOPT_INCLUDE_DIRS})
    add_definitions(${IPOPT_DEFINITIONS})
 endif()
 
 add_definitions(-D_USE_MATH_DEFINES)
 add_library(${PROJECT_NAME} ${folder_source} ${folder_header})
+
+if(ICUB_USE_IPOPT)
+  target_include_directories(${PROJECT_NAME} PUBLIC  "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"
+                                                     "$<INSTALL_INTERFACE:$<INSTALL_PREFIX>/${CMAKE_INSTALL_INCLUDEDIR}>"
+                                             PRIVATE ${IPOPT_INCLUDE_DIRS})
+else()
+  target_include_directories(${PROJECT_NAME} PUBLIC  "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"
+                                                     "$<INSTALL_INTERFACE:$<INSTALL_PREFIX>/${CMAKE_INSTALL_INCLUDEDIR}>")
+endif()
```

Comment:
Calls to `target_include_directories` can also be separated, so you can also write: 
~~~cmake
target_include_directories(${PROJECT_NAME} PUBLIC  "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"
                                                    "$<INSTALL_INTERFACE:$<INSTALL_PREFIX>/${CMAKE_INSTALL_INCLUDEDIR}>")
                                                    
if(ICUB_USE_IPOPT)
    target_include_directories(${PROJECT_NAME}  PRIVATE ${IPOPT_INCLUDE_DIRS})
endif() 
~~~